assignments/this.js

```diff
@@ -1,26 +1,97 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. Window/Global Object Binding
-* 2. Implicit Binding
-* 3. New binding
-* 4. Explicit binding
-*
+* 1. Window/Global Object Binding - "this" is automatically set to the window or console.
+* 2. Implicit Binding - "this" is provided implicitly by a preceding ".", then followed by a predefined method.
+* 3. New binding - "this" is provided by a constructor function using the keyword "new".
+* 4. Explicit binding - "this" is provided explicitly by means of a built-in ".call" and/or "apply" method.
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
-// code example for Window Binding
+function sayName(name){
+    console.log(this);
+    return name;
+}
+sayName('Saipher');
 
 // Principle 2
 
-// code example for Implicit Binding
+const zureti = {
+    name: "Zureti",
+    say: function(){ console.log(`My last name is ${this.name}.`)}
+}
+
+zureti.say();
 
 // Principle 3
 
-// code example for New Binding
+//Tried to create this function. It worked at first until I made a mistake and tried to fix it but wasn't able to. I'll revisit later.
+// function Friend(person){
+//     this.name = person.name;
+//     this.introduction = person.introduction;
+//     this.speak = function(){
+//         console.log(this.introduction + this.person);
+//         console.log(this);
+
+//         let inati = new Friend({
+//             name: "Umi",
+//             introduction: "Nice to meet you,"
+//         });
+        
+//         let umi = new Friend({
+//             name: "Inati",
+//             introduction: "Nice to meet you,"
+//         });
+//     }
+// }
+    // this.name = person.name;
+    // this.introduction = "Nice to meet you.";}
+
+
+//  inati = new friend('Umi');
+//  umi = new friend('Inati');
+
+// inati.speak();
```

Comment:
if you want to review this let me know, the issue probably began when you made the new friend inside of the function